### ISSUE TITLE
-Jz on 'flat' modules

### ISSUE THREAD

**Issue by joa-quim** [GenericMappingTools/gmt#123]
I think I (partially) brought this before thought not in public. It always puzzles me when I see _flat_ modules listing **-Jz** in the options list. This strikes as a contradiction. If they are _flat_ why should we might want to use the **-Jz** option? OK, by _flat_ I mean those modules that create flat images (or no images at all). I found these in that category
```
gmtlogo
psclip
psmask
pshistogram
psmask
pstext
pswiggle
grdimage
psimage
```
than looked in our test suite and only these two, ``sphere_volume`` (uses **-Jz** in a ``pstext`` call) and ``3dbars`` (**-JZ** in ``grdimage``) use it. Removing **-Jz** from the former makes no difference and the test still passes (meaning that its use is useless). Regarding the ``3dbars``, it fails if we remove it but that happens because the **-JZ** and **-BWSNEZ+b** are used to plot and annotate the 3D cube. However, we can pass those instructions to the following ``psxyz`` module, where they make more sense.

Right, so the **-Jz** is useful if we want to draw and annotate a _z_ axis but we will never(?) need it if we are not plotting any data that uses the 3rd dim, and in that case we can always control the _z_ axis from that module. So, am I missing anything that prevents us to just remove **-Jz** from all _flat_ modules?

**Comment by PaulWessel** [comment]
I understand what you mean by flat (no grdview) but some of these I think consider a z-coordinates and then -R and -Jz do enter into what level we plot the flat stuff at, no?

**Comment by joa-quim** [comment]
I first thought that this could be just removed in GMT6 that allows breakages, though this one would be a very minor one. And I sill do but it requires more changes. As is, if the testing machinery finds a ``-R1/2/3/4/5/6`` it will error if no ``-Jz`` is present. This implies that a ``-Jz`` becomes mandatory if we have a solitary ``-R`` that follows a previous ``-R`` setting with all 6 params. This is what actually happens in the ``psxyz/3dbars.sh`` test. But I think this is the only reason why we may have to consider z-coordinates in the _flat_ modules.

We should be able to solve this easily if only ``grdview`, psxyz``and ``psbasemap`` use, _and test_, ``-Jz``. With this and for backward compat reasons we could still accept, and ignore, ``-Jz`` in all _flats_ and make clear in docs that in GMT6 the z-axis is only configurable in ``grdview, psxyz`` and ``psbasemap``

**Comment by PaulWessel** [comment]
Time to close, @joa-quim ?

**Comment by joa-quim** [comment]
Solved in #860
